### PR TITLE
feat: add vitest

### PR DESCRIPTION
**Pull request by anbraten**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          fetch-depth: 0
+
+      - name: Set node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
+      - run: npx changelogithub
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
```
Comment (bdbch) [inline .github/workflows/release.yml:26]: If the build branch is done and we have a place to publish we could add building and publishing here too - cool stuff!

**Comment by anbraten** [comment]
Would go with some low value like `0.0.1` and let semver bump it. We could do as many breaking changes as needed in `0.`

**Comment by MathurAditya724** [comment]
Yeah, based upon our previous discussions, we will stick to the patch version only, once we have something that everyone can use we can start rolling out from minor.

**Comment by bdbch** [comment]
Alright! Feel free to throw my alpha-stuff out the window 🧨

**Pull request merged**